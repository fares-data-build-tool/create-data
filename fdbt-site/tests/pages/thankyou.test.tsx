import * as React from 'react';
import { shallow } from 'enzyme';
import Thankyou, { getServerSideProps } from '../../src/pages/thankyou';
import { getMockContext } from '../testData/mockData';

describe('pages', () => {
    describe('thankyou', () => {
        it('should render correctly', () => {
            const tree = shallow(<Thankyou uuid="a1b2c3d4e5" />);
            expect(tree).toMatchSnapshot();
        });
    });

    describe('getServerSideProps', () => {
        it('retrieves the uuid correctly', () => {
            const ctx = getMockContext({
                cookies: {},
                body: null,
                uuid: { operatorUuid: '84c7b1b4-e178-4849-bc49-bd32cdb2db39' },
            });

            const expectedResults = {
                props: {
                    uuid: '84c7b1b4-e178-4849-bc49-bd32cdb2db39',
                },
            };

            const results = getServerSideProps(ctx);
            expect(results).toEqual(expectedResults);
        });
    });
});
