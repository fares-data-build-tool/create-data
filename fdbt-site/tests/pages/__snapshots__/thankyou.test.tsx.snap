// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages thankyou should render correctly 1`] = `
<Component
  description="Thank you page for the Fares Data Build Tool"
  title="Thank You - Fares Data Build Tool"
>
  <div
    className="govuk-panel govuk-panel--confirmation"
  >
    <h1
      className="govuk-panel__title"
      id="thank-you-page-heading"
    >
      Upload complete
    </h1>
    <div
      className="govuk-panel__body"
      id="uuid-ref-number"
    >
      Your reference number
      <br />
      <strong>
        a1b2c3d4e5
      </strong>
    </div>
  </div>
  <h2
    className="govuk-heading-m"
  >
    What happens next
  </h2>
  <p
    className="govuk-body"
  >
    Thank you for submitting your fares data.
  </p>
  <p
    className="govuk-body"
  >
    Your data will be converted to the NeTEx format and published to the open data hub. You will be contacted should there be a problem with your upload.
  </p>
  <p
    className="govuk-body"
  >
    If you would like to provide feedback on this service, please click
     
    <a
      className="govuk-link"
      href="mailto:fdbt@transportforthenorth.com?bcc=tfn-feedback@infinityworks.com"
    >
      here
    </a>
  </p>
  <br />
  <a
    className="govuk-button"
    data-module="govuk-button"
    draggable="false"
    href="/"
    role="button"
  >
    Add another fare
  </a>
</Component>
`;
